src/libs/OptionsListUtils.ts

```diff
@@ -1320,12 +1350,63 @@ function isReportSelected(reportOption: ReportUtils.OptionData, selectedOptions:
     return selectedOptions.some((option) => (option.accountID && option.accountID === reportOption.accountID) || (option.reportID && option.reportID === reportOption.reportID));
 }
 
+function createOptionList(personalDetails: OnyxEntry<PersonalDetailsList>, reports?: OnyxCollection<Report>) {
+    const reportMapForAccountIDs: Record<number, Report> = {};
+    const allReportOptions: Array<SearchOption<Report>> = [];
+
+    if (reports) {
+        Object.values(reports).forEach((report) => {
+            if (!report) {
+                return;
+            }
+
+            const isSelfDM = ReportUtils.isSelfDM(report);
+            // Currently, currentUser is not included in visibleChatMemberAccountIDs, so for selfDM we need to add the currentUser as participants.
```

Comment:
are we planning on changing this? In what case do we care about the selfDM accountID

Reply:
Hard to say, those changes were originally made in #37165, I just extracted it to separate function